tests/tests_integration/test_api/test_data_workflows.py

```diff
@@ -291,138 +282,115 @@ def workflow_data_modeling_trigger(cognite_client: CogniteClient, add_multiply_w
-                                    external_id="integration_test-workflow_definitions-test_create_delete-task1"
-                                    + random_string(5),
-                                    parameters=FunctionTaskParameters(
-                                        external_id="integration_test-workflow_definitions-test_create_delete-task1-function",
-                                        data={"a": 1, "b": 2},
-                                    ),
-                                )
-                            ]
-                        ),
-                    )
-                ],
-                description="This is ephemeral workflow definition for testing purposes",
-            ),
-        )
-        cognite_client.workflows.versions.delete(version.as_id(), ignore_unknown_ids=True)
-
-        created_version: WorkflowVersion | None = None
-        try:
-            created_version = cognite_client.workflows.versions.upsert(version)
-
-            assert created_version.workflow_external_id == version.workflow_external_id
-            assert created_version.workflow_definition.description == version.workflow_definition.description
-            assert isinstance(created_version.workflow_definition.hash_, str)
-        finally:
-            if created_version is not None:
-                cognite_client.workflows.versions.delete(
-                    created_version.as_id(),
-                )
-                cognite_client.workflows.delete(created_version.workflow_external_id)
+    def test_upsert_preexisting(self, cognite_client: CogniteClient, new_workflow_version: WorkflowVersion) -> None:
+        new_workflow_version.workflow_definition.description = "Updated description for testing purposes"
+        updated_version = cognite_client.workflows.versions.upsert(new_workflow_version.as_write())
+
+        assert updated_version.workflow_external_id == new_workflow_version.workflow_external_id
+        assert updated_version.version == new_workflow_version.version
+        assert updated_version.workflow_definition.description == new_workflow_version.workflow_definition.description
 
     def test_list_workflow_versions(
         self, cognite_client: CogniteClient, workflow_version_list: WorkflowVersionList
     ) -> None:
-        listed = cognite_client.workflows.versions.list(workflow_version_list.as_ids())
+        listed_by_wf_xid = cognite_client.workflows.versions.list(workflow_version_list[0].workflow_external_id)
+        listed_by_wf_version_id = cognite_client.workflows.versions.list(
+            WorkflowVersionId(workflow_version_list[0].workflow_external_id)
+        )
+        listed_by_as_ids = cognite_client.workflows.versions.list(workflow_version_list.as_ids())
 
-        assert len(listed) == len(workflow_version_list)
-        assert listed == workflow_version_list
+        ids_tuples = [tuple(x.__dict__.values()) for x in workflow_version_list.as_ids()]
```

Comment:
There seem to be an `as_primitive` method you can use:
```suggestion
        ids_tuples = [wid.as_primitive() for wid in workflow_version_list.as_ids()]
```